rsc/src/main/scala/rsc/lexis/Keywords.scala

```diff
@@ -2,58 +2,62 @@
 // Licensed under the Apache License, Version 2.0 (see LICENSE.md).
 package rsc.lexis
 
+import java.util.HashMap
+
 trait Keywords {
-  val keywords: Map[String, Token] = Map(
-    "abstract" -> ABSTRACT,
-    "case" -> CASE,
-    "catch" -> CATCH,
-    "class" -> CLASS,
-    "def" -> DEF,
-    "do" -> DO,
-    "else" -> ELSE,
-    "extends" -> EXTENDS,
-    "false" -> FALSE,
-    "final" -> FINAL,
-    "finally" -> FINALLY,
-    "for" -> FOR,
-    "forSome" -> FORSOME,
-    "if" -> IF,
-    "implicit" -> IMPLICIT,
-    "import" -> IMPORT,
-    "lazy" -> LAZY,
-    "match" -> MATCH,
-    "new" -> NEW,
-    "null" -> NULL,
-    "object" -> OBJECT,
-    "override" -> OVERRIDE,
-    "package" -> PACKAGE,
-    "private" -> PRIVATE,
-    "protected" -> PROTECTED,
-    "return" -> RETURN,
-    "sealed" -> SEALED,
-    "super" -> SUPER,
-    "this" -> THIS,
-    "throw" -> THROW,
-    "trait" -> TRAIT,
-    "try" -> TRY,
-    "true" -> TRUE,
-    "type" -> TYPE,
-    "val" -> VAL,
-    "var" -> VAR,
-    "while" -> WHILE,
-    "with" -> WITH,
-    "yield" -> YIELD,
-    "_" -> USCORE,
-    ":" -> COLON,
-    "=" -> EQUALS,
-    "=>" -> ARROW,
-    "⇒" -> ARROW,
-    "<-" -> LARROW,
-    "←" -> LARROW,
-    "<:" -> SUBTYPE,
-    "<%" -> CATCH,
-    ">:" -> SUPERTYPE,
-    "#" -> HASH,
-    "@" -> AT
-  )
+  val keywords = {
```

Comment:
I think this should be moved to the companion - the map build will be per-run rather an an initialisation cost.

Reply:
Can you elaborate?

Reply:
Of course. 
The val will be recomputed each time you instantiate an implementing class.  
Although when I went to look I realised its on the package object, so my point is moot.